system-addon/content-src/components/Sections/Sections.jsx

```diff
@@ -105,6 +110,20 @@ class Section extends React.PureComponent {
     }
   }
 
+  needsImpressionStats(cards) {
+    if (!this.impressionCardGuids) {
+      return true;
+    }
+
+    for (let i = 0; i < cards.length; i++) {
+      if (cards[i].guid !== this.impressionCardGuids[i]) {
```

Comment:
nit: this works assuming that `cards` and `impressionCardGuids` are of the same length, we can add a short-circuit before this foo loop to avoid the potential index overflow bug,
```
if (!this.impressionCardGuids || this.impressionCardGuids.length !== cards.length) {
  return true;
}
```  
Since cards are coming from 3rd party, and may come with any possible size, just be safe :)

Reply:
@ncloudioj thanks, fixed.